### PR TITLE
Add GCN compiler target

### PR DESCRIPTION
**Pull request by jpsamaroo**
AMDGPUnative PR to follow

### CODE REVIEW

**Comment by jpsamaroo** [comment]
The first two commits are from #1

**Comment by jpsamaroo** [comment]
AMDGPUnative PR: https://github.com/JuliaGPU/AMDGPUnative.jl/pull/46

**Comment by maleadt** [comment]
Rebased. Would be good to add some local tests + reverse CI.

File: src/amdgpu.jl [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    feat = ""
+    optlevel = LLVM.API.LLVMCodeGenLevelDefault
+    reloc = LLVM.API.LLVMRelocPIC
+    tm = TargetMachine(t, triple, cpu, feat, optlevel, reloc)
+    asm_verbosity!(tm, true)
+
+    return tm
+end
+
+llvm_datalayout(::AMDGPUCompilerTarget) =
```
Comment (maleadt) [inline src/amdgpu.jl:37]: You only need to override the datalayout if the one selected by LLVM isn't the correct one -- chances are it is.

**Comment by jpsamaroo** [comment]
@maleadt aside from the 1.4 failures, there are two broken testsets that I've marked. The invalid addrspacecast failure is expected (to/from Julia's addrspaces), the other I don't recognize.

**Comment by maleadt** [comment]
Looking at the segfault, I suspected a violated LLVM assertions, so I added a CI build for that:
```
 julia: /workspace/srcdir/llvm-project/llvm/lib/IR/Instructions.cpp:3262: llvm::BitCastInst::BitCastInst(llvm::Value*, llvm::Type*, const llvm::Twine&, llvm::Instruction*): Assertion `castIsValid(getOpcode(), S, Ty) && "Illegal BitCast"' failed.
```

**Comment by maleadt** [comment]
For tests that are only broken on 1.4, you can import and use `@test_broken_if` from https://github.com/JuliaGPU/CUDAnative.jl/blob/7b8973371edea0b04900cbb086e9ad35d189db4c/test/util.jl#L105-L114

**Pull request merged**